scripts/stealthy.js

```diff
@@ -40,6 +41,31 @@ export class Stealthy {
           // idea that active skills need to win outright to change the status quo. Passive
           // perception means that stealth is being the active skill.
           let perception = spot?.flags.stealthy?.spot ?? (source.system.skills.prc.passive + 1);
+          
+          // check target Token Lighting conditions via effects usage
+          // look for effects that indicate Dim or Dark condition on the token
+          const  tokenLight = game.settings.get('stealthy', 'tokenLighting');
+          if (tokenLight){
+            let lightLevel = 2; // bright
+            if (target?.effects.find(e => e.label === 'Dark' && !e.disabled)){lightLevel=0;}
+            if (target?.effects.find(e => e.label === 'Dim' && !e.disabled)){lightLevel=1;}
+            let lightLevelText = Stealthy.lightNumTable[lightLevel];
+            Stealthy.log("testVisionStealth:", lightLevelText);
+
+            // check if Darkvision is in use, bump light level accordingly
+            if (visionSource.visionMode?.id === 'darkvision'){lightLevel = lightLevel + 1;}
+            lightLevelText = Stealthy.lightNumTable[lightLevel];
+            Stealthy.log("post lightLevel:", lightLevelText);
+
+            // adjust passive perception depending on light conditions of target token
+            // don't adjust for active perception checks via 'spot' flag usage
+            Stealthy.log("Pass prc:",perception);
+            if (lightLevel<2 && !spot?.flags.stealthy?.spot){
+              perception=perception-5;
+              Stealthy.log("Post Pass prc:",perception);
```

Comment:
Can you collapse all these logs into a single one before, inside or after the if? This is a lot of spew line, if they were on one line at least it would be better. For things checked in, I'd stick with logging only when you affect something like dropping the perception

Reply:
I did this on my branch ```          if (perception <= stealth) {
            Stealthy.log(`${visionSource.object.name}'s ${perception} didn't beat ${config.object.name}'s ${stealth}`);
            return false;
          }
```

Reply:
Logging collapse to a single data structure